lib/mkdir-file.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+var File = require('winston-uber').transports.File;
+var fs = require('fs');
+var mkdirp = require('mkdirp');
+var util = require('util');
+
+function MkdirFile() {
+    File.apply(this, arguments);
+}
+
+util.inherits(MkdirFile, File);
+
+MkdirFile.prototype.name = 'MkdirFile';
+
+MkdirFile.prototype.open = function open(cb) {
+    // Same as File#open
+    if (this.opening) {
+        return cb(true);
+    }
+
+    if (!this._stream || (this.maxsize && this._size >= this.maxsize)) {
+        // do proper stat
+        return this._open(cb);
+    }
+
+    // By default _stream means we are open
+    return cb();
+};
+
+MkdirFile.prototype._open = function _open(cb) {
+    var self = this;
+    var dirname = self.dirname;
+
+    fs.exists(dirname, function (exists) {
```

Comment:
Classic race condition here, if you haven’t arranged to avoid contention otherwise. Just try to mkdirp, then try to open. Handle the errors gracefully. Use the right flags. Don’t forget the "x" flag if you need it. https://nodejs.org/api/fs.html#fs_fs_open_path_flags_mode_callback